wordpress-mobile/WordPress-Android
# Jetpack Focus: Hook up Jetpack feature card CTA interactions

Parent #17336 

This PR implements the CTA interactions on the Jetpack Feature Card 
Includes
- On card click navigation event
- On Hide Me click - refresh the UI
- On Remind Later click - refresh the UI
- On Learn More click - navigate to external browser and launch URL
- Hooks in the call to remote config to get the value of `phase_three_blog_post` and tacks on the host "https://" if it doesn't exist otherwise the launch will fail.
- Sets the visibility of Learn More link using the value of the URL

**Note**: The overlay feature view is not shown on card click; waiting on PR #17580 to merge before that work can start.

**Merge Instructions**
~1 - Ensure that #17681 and #17692 have been merged~
~2 - Remove "Not ready for merge" label~
3 - Merge as normal

**To test:**
Test 1: Learn More click launches external browser and removes the card from the menu tab
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Privacy Settings and turn on Collection Information
- Navigate to -> Debug Settings -> and  Enable `jp_removal_three` and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown
- Tap on the More "horizontal ellipse" to launch the more menu
- Tap on the `Remind me later` item
- ✅ Verify the view refreshes and the Jetpack Feature card is no longer shown
- Swipe the app closed
- Relaunch the app
- ✅ Verify the Jetpack Feature card is no longer shown
- Logout to clear the app prefs

Test 2: Hide this removes the card from the menu tab
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Privacy Settings and turn on Collection Information
- Navigate to -> Debug Settings -> and  Enable `jp_removal_three` and restart the app *Note: this value remains across log outs
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown
- Tap on the More "horizontal ellipse" to launch the more menu
- Tap on the `Hide this` item
- ✅ Verify the view refreshes and the Jetpack Feature card is no longer shown
- Swipe the app closed
- Relaunch the app
- ✅ Verify the Jetpack Feature card is no longer shown
- Logout to clear the app prefs

Test 3: Tapping Learn More launches external browser
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Privacy Settings and turn on Collection Information
- Navigate to -> Debug Settings -> and  Enable `jp_removal_three` and restart the app *Note: this value remains across log outs
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown
- Tap on the Learn More link
- ✅ Verify an external browser launches and you land on the URL (jetpack.com)
- Logout to clear the app prefs


## Regression Notes
1. Potential unintended areas of impact
Jetpack feature card shows when it shouldn't

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
